### ISSUE TITLE
Search the book for outdated terms and modernise them

### ISSUE THREAD

**Issue by sgibson91** [alan-turing-institute/the-turing-way#2149]
### Summary

This is a follow-up issue to https://github.com/alan-turing-institute/the-turing-way/issues/1210

Now that our default branch is called `main` we should do a search for other such terms (e.g. `blacklist` and `whitelist`) and update them with more modern alternatives, (e.g. `denylist` and `allowlist`).

### What needs to be done?

- [ ] Curate a list of terms to avoid and appropriate alternatives (Is there a catalogue already in existence we can link to?)
- [ ] Perform a search and replace on the book (and other repo contents) following the previously defined guidance

### Who can help?

- Anyone!

**Comment by aldenc** [comment]
I have noted several references to the "master" branch in the text. I will create PRs to change those to "main" and link them here.

**Comment by sgibson91** [comment]
Hey @aldenc! Just a note to say that the majority of the instances of "master" have already been fixed _in the source code_ but the _rendered book_ hasn't been updated yet due to this bug https://github.com/alan-turing-institute/the-turing-way/issues/2150 We're waiting on someone with admin access to netlify to rectify it :D

**Comment by sgibson91** [comment]
#2150 is now rectified so you see be able to see the up-to-date book now :)

**Comment by sgibson91** [comment]
I just found a bot that can do this! https://probot.github.io/apps/in-solidarity/

**Comment by aldenc** [comment]
The bot has been tested and is now running on The Turing Way repo ([PR 2237](https://github.com/alan-turing-institute/the-turing-way/pull/2237)), and rules are configured as follows:

rule | level
-- | --
master | failure
slave | failure
whitelist | failure
blacklist | failure

The bot will run a check on any new PRs (it [parses gitdiff](https://github.com/jpoehnelt/in-solidarity-bot/blob/main/src/parse.ts)) and the check will fail if any of those terms appear. The configuration file is located at .github/in-solidarity.yml and we can update it to create new rules as needed.

**Comment by sgibson91** [comment]
The developer of the bot is also interested in figuring out how to open new issues when undesired terms are found https://github.com/jpoehnelt/in-solidarity-bot/issues/72 It might be a nice opportunity for a TTW member with JavaScript experience to contribute upstream

**Comment by edwardabraham** [comment]
Thanks for raising this. Just noting that the diagrams on this page still need to be updated to refer to the main branch: https://the-turing-way.netlify.app/reproducible-research/vcs/vcs-workflow.html

**Comment by aldenc** [comment]
closing this issue since all comments have been addressed
